analyses/ewings-cell-type/03-copykat.Rmd

```diff
@@ -0,0 +1,433 @@
+- Calculate the confusion matrix comparing manual annotations of tumor cells to using `CopyKAT` to annotate tumor cells. 
+- Compare the annotations from `CopyKAT` to cell type annotations from `SingleR` and `CellAssign`. 
+
+
+## Setup
+
+```{r packages}
+suppressPackageStartupMessages({
+  # load required packages
+  library(SingleCellExperiment)
+  library(ggplot2)
+  library(copykat)
+})
+```
+
+
+```{r base paths}
+# The base path for the OpenScPCA repository, found by its (hidden) .git directory
+repository_base <- rprojroot::find_root(rprojroot::is_git_root)
+
+# The current data directory, found within the repository base directory
+data_dir <- file.path(repository_base, "data", "current")
+sample_dir <- file.path(data_dir, "SCPCP000015", params$sample_id)
+
+# The path to this module
+module_base <- file.path(repository_base, "analyses", "ewings-cell-type")
+```
+
+
+```{r}
+# source in helper functions for make_jaccard_matrix() and jaccard()
+jaccard_functions <- file.path(module_base, "scripts", "utils", "jaccard-functions.R")
+source(jaccard_functions)
+```
+
+
+```{r paths}
+# Input files
+sce_filename <- glue::glue("{params$library_id}_processed.rds")
+sce_file <- file.path(sample_dir, sce_filename)
+
+# tumor/normal classifications 
+classifications_filename <- glue::glue("{params$library_id}_tumor_normal_classifications.tsv")
+classifications_file <- file.path(module_base, "results", "marker_gene_analysis", classifications_filename)
+
+# copykat results 
+results_dir <- file.path(module_base, "results", "copykat", params$library_id)
+
+# predictions files 
+predictions_file <- glue::glue("{params$library_id}_copykat_prediction.txt")
+
+predictions_paths <- c(
+  no_ref = file.path(results_dir, "no_normal", predictions_file),
+  with_ref = file.path(results_dir, "endothelial_normal", predictions_file)
+)
+
+# full results gene by cell 
+full_ck_result_file <- glue::glue("{params$library_id}_copykat_CNA_raw_results_gene_by_cell.txt")
+  
+full_ck_result_paths <- c(
+  no_ref = file.path(results_dir, "no_normal", full_ck_result_file),
+  with_ref = file.path(results_dir, "endothelial_normal", full_ck_result_file)
+)
+
+```
+
+
+```{r}
+# read in processed sce
+sce <- readr::read_rds(sce_file)
+
+# read in tumor normal classifications 
+manual_classifications_df <- readr::read_tsv(classifications_file)
+
+# read in ck predictions from both reference types (no_normal and with_normal)
+ck_results_df <- predictions_paths |> 
+  purrr::map(readr::read_tsv) |> 
+  dplyr::bind_rows(.id = "reference_used")
+
+# read in full gene by cell copy number detection results 
+full_ck_results_df <- full_ck_result_paths |> 
+  purrr::map(readr::read_tsv) |> 
+  dplyr::bind_rows(.id = "reference_used")
+```
+
+
+## CopyKAT results
+
+Below we look at the heatmaps produced by `CopyKAT`. 
+
+### Heatmap without reference
+
+![](`r file.path(results_dir, "no_normal", "SCPCL000822_copykat_heatmap.jpeg")`)
```

Comment:
Yea I totally agree! I did try to create the plot on my own but it's so large it crashes. I think it would be more beneficial to be able to label isolate specific aneuploides and then plot those, but definitely not worth the effort at this stage. You'll see that the information from InferCNV I think actually does a better job of giving us individual CNV annotation information (and more informative plots).